src/main/java/com/library/service/ValidateService.java

```diff
@@ -0,0 +1,18 @@
+package com.library.service;
+
+import com.library.model.entity.ParentEntity;
+import org.springframework.stereotype.Component;
+
+import java.security.InvalidParameterException;
+
+@Component
+public interface ValidateService<T extends ParentEntity> {
```

Comment:
Nowadays it is outdated approuch of validation.

Read about data validation in spring app and about filters in rest app.

And here and in other places please remove such validation by hand.